libp2p-xtra/src/node.rs

```diff
@@ -0,0 +1,508 @@
+use crate::multiaddress_ext::MultiaddrExt as _;
+use crate::upgrade;
+use crate::Connection;
+use crate::Substream;
+use anyhow::bail;
+use anyhow::Context as _;
+use anyhow::Result;
+use async_trait::async_trait;
+use futures::future::BoxFuture;
+use futures::stream::BoxStream;
+use futures::AsyncRead;
+use futures::AsyncWrite;
+use futures::TryStreamExt;
+use libp2p_core::identity::Keypair;
+use libp2p_core::transport::Boxed;
+use libp2p_core::transport::ListenerEvent;
+use libp2p_core::Multiaddr;
+use libp2p_core::PeerId;
+use libp2p_core::Transport;
+use multistream_select::NegotiationError;
+use multistream_select::Version;
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::marker::PhantomData;
+use std::time::Duration;
+use thiserror::Error;
+use tokio_tasks::Tasks;
+use xtra::message_channel::StrongMessageChannel;
+use xtra_productivity::xtra_productivity;
+use xtras::SendAsyncSafe;
+
+/// An actor for managing multiplexed connections over a given transport.
+///
+/// The actor does not inflict any policy on connection and/or protocol management.
```

Comment:
🙃 I think "inflict" doesn't quite fit here. We could for example use:

```suggestion
/// The actor does not impose any policy on connection and/or protocol management.
```